code/modules/projectiles/guns/energy/special.dm

```diff
@@ -113,26 +111,19 @@
 	origin_tech = "combat=4;powerstorage=3"
 
 /obj/item/weapon/gun/energy/kinetic_accelerator/shoot_live_shot()
-	overheat = 1
-	spawn(overheat_time)
-		overheat = 0
-		recent_reload = 0
 	..()
+	addtimer(src, "reload", overheat_time)

```

Comment:
If you ultimately decide to go with a timer anyway (I personally don't recommend it), I suggest putting this *before* the parent is called. Why? If there's ever a change to the fire delay, I believe this may end up making the timer longer than you intend (I could be wrong on this)